packages/effect/src/ReadonlyRecord.ts

```diff
@@ -164,7 +168,9 @@ export const fromIterableBy = <A>(items: Iterable<A>, f: (a: A) => string): Reco
  * @category conversions
  * @since 2.0.0
  */
-export const fromEntries: <A>(self: Iterable<readonly [string, A]>) => Record<string, A> = fromIterableWith(identity)
+export const fromEntries: <K extends string, A>(self: Iterable<readonly [K, A]>) => Record<K, A> = fromIterableWith(
```

Comment:
all the `from*` APIs 
- `fromIterableWith`
- `fromIterable`
- `fromIterableBy`
- `fromEntries`
are unsafe now (`Record<K, A>` where `K` is a union of literals, let's say "a" | "b", is equivalente to a struct `{ a: value1, b: value2 }`).

Example with `fromEntries`:

```ts
const myiterable: Iterable<["a" | "b", string]> = []
// const myrecord: Record<"a" | "b", string>
const myrecord = fromEntries(myiterable)
myrecord.a.trim() // throws
```